internal/v1/handler.go

```diff
@@ -298,6 +291,26 @@ func (h *Handlers) GetComposeStatus(ctx echo.Context, composeId uuid.UUID) error
 	return ctx.JSON(http.StatusOK, status)
 }
 
+func (h *Handlers) handleNotFoundResponse(resp *http.Response) error {
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to read response body").SetInternal(err)
+	}
+	// Composes can get deleted in composer, usually when the image is expired
+	return echo.NewHTTPError(http.StatusNotFound, string(body))
+}
+
+func (h *Handlers) handleErrorResponse(ctx echo.Context, resp *http.Response) error {
+	httpError := echo.NewHTTPError(http.StatusInternalServerError, "Failed querying compose status")
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		ctx.Logger().Errorf("Unable to parse composer's compose response: %v", err)
```

Comment:
I know this is a pre-existing string but it's a bit misleading, here it's not about `parse` but about reading.

Reply:
AFAICT this still an open topic, no? Maybe: `ctx.Logger().Errorf("Unable to read error response data: %v", err)` ?